README.md

```diff
@@ -775,3 +774,202 @@ resource "google_compute_global_forwarding_rule" "app" {
+```bash
+source ~/your_repo/ansible/venv/bin/activate
+```
+
+* After the previous paragraph, all software will be installed in a separate virtual environment `venv` and this software will not be available outside of this environment. (To disable the virtual environment, run the `deactivate` command)
+* Create file `~/your_repo/ansible/requirements.txt`. This file will specify the necessary programs and the required version to work with. In this file add next content:
+
+```text
+  ansible==2.10.6
+  molecule>=2.6
+  pytest-testinfra>=1.10
+  python-vagrant>=0.5.15
+  molecule_vagrant==1.0.0 
+  flake8>=6.0.0
+  ansible-lint>=6.10.0
+```
+
+* Activate your venv and execute command, which installed pre-requesits soft from your file:
+
+```bash
+  python3 -m pip install -r ~/your_repo/ansible/requirements.txt
+```
+
+* Install Vagrant and VirtualBox
+  
+```bash
+  
+wget -O- https://apt.releases.hashicorp.com/gpg | gpg --dearmor | sudo tee /usr/share/keyrings/hashicorp-archive-keyring.gpg
+echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/hashicorp.list
+sudo apt update && sudo apt install vagrant
+
+echo "deb [arch=amd64 signed-by=/usr/share/keyrings/oracle-virtualbox-2016.gpg] https://download.virtualbox.org/virtualbox/debian bionic contrib" | sudo tee -a /etc/apt/sources.list.d/virtualbox.list
+wget -O- https://www.virtualbox.org/download/oracle_vbox_2016.asc | sudo gpg --dearmor --yes --output /usr/share/keyrings/oracle-virtualbox-2016.gpg
+sudo apt update
+sudo apt install virtualbox-6.1
+```
+
+* Install Ruby. Command:
+
+```bash
+  sudo apt-get install ruby-full
+ ```
+
+### Setup Ansible
+
+* In directory `~/your_repo/ansible` create next directories
+  
+```text
+ environments
+ gce_py
+ playbooks
+ roles
+```
+
+and files
+
+```text
+ ansible.cfg
+ Vagrantfile
+ ```
+
+* In `ansible.cfg` there should be the following content
+
+```text
+[defaults]
+interpreter_python = auto
+inventory = environments/stage/gce.py  # what kind of environment should playbooks work with (stage, prod, dev and etc)
+roles_path = ./roles  # location of roles
+remote_user = appuser
+private_key_file = path to your ssh key
+host_key_checking = False
+retry_files_enabled = False
+vault_password_file = path to your vault key word
+
+[diff]
+always = True
+context = 5
+```
+
+* In `Vagrantfile` there should be config for your local VMs (see file in repository)
+
+* Create GCE Service Account <https://cloud.google.com/compute/docs/access/service-accounts> and save on your local host in directory `~/your_repo/ansible/gce_py/` key in `*.json` format from GCP for your dynamic inventory script
+  
+* In directory `environments` contains files and folders responsible for various environments, which contain their own variables and settings, as well as a dynamic inventory script. Your roles will take values from these directories depending on the environment
```

Comment:
`contains files and folders responsible` -> `place files and folders responsible`